torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -108,8 +108,29 @@ static c10::optional<c10::ScalarType> InferExpectedScalarType(const Node* n) {
           // which is by default considered as a tensor.
           typesFromScalars.emplace_back(c10::kLong);
         } else if (nkind == onnx::Constant) {
-          typesFromScalars.emplace_back(
-              input->node()->t(attr::value).scalar_type());
+          auto tensor = input->node()->t(attr::value);
+          auto rank = tensor.dim();
+          auto scalar_type = tensor.scalar_type();
+          // Mimic PyTorch scalar type promotion logic
+          // from https://github.com/pytorch/pytorch/issues/9515
+          if (rank == 0) {
+            auto default_scalar_type =
+                at::typeMetaToScalarType(at::get_default_dtype());
+            switch (scalar_type) {
+              case at::kDouble:
+                typesFromScalars.emplace_back(default_scalar_type);
+                break;
+              case at::kLong:
```

Comment:
Can you specify the behavior here for case Long, maybe in comments?

Reply:
Added some comments.